### ISSUE TITLE
Qutebrowser by default takes very long time (43 seconds) to open a link

### ISSUE THREAD

**Issue by Hi-Angel** [qutebrowser/qutebrowser#6813]
**Version info**:

```
qutebrowser v2.4.0
Git commit:
Backend: QtWebEngine 5.15.6, based on Chromium 87.0.4280.144
Qt: 5.15.2
```

**Does the bug happen if you start with `--temp-basedir`?**:

Yes

**Description**

This report stems [from this discussion](https://github.com/qutebrowser/qutebrowser/issues/6787#issuecomment-955559395). Upon clicking a link somewhere in the system, thus making Qutebrowser to open the link, Qutebrowser may take as much as 43 seconds to react, given cold filesystem cache. It may be instantaneous if cache is hot though.

There was long time [a workaround for this problem](https://github.com/qutebrowser/qutebrowser/blob/master/scripts/open_url_in_instance.sh), but since #6787 it stopped working. In discussion ensued it turned out that such long time opening a link is a bug, so here's a report.

**How to reproduce**

1. Open Qutebrowser
2. Execute `echo 1 > /proc/sys/vm/drop_caches` to drop file system caches.
3. Execute `time xdg-open https://example.com` to open the link in Qutebrowser

**Logs**

* [newly-launched.log](https://github.com/qutebrowser/qutebrowser/files/7447284/newly-launched.log): it is dropping the caches, then running `qutebrowser --debug --basedir=/tmp/foo https://example.com &>/tmp/newly-launched.log`.
    After qutebrowser window appeared I ^Ced it to stop logging.
* [existing-process.log](https://github.com/qutebrowser/qutebrowser/files/7447292/existing-process.log): this is launching process as above, then dropping caches, then `qutebrowser --debug --basedir=/tmp/foo https://example.com &>/tmp/existing-process.log`

**Comment by The-Compiler** [comment]
In your `existing-process` log, the main delay is in two places:

```
20:37:04 DEBUG    init       earlyinit:init_log:272 Log initialized.
20:37:32 DEBUG    init       app:run:85 Main process PID: 95253
```

and:

```
20:37:33 DEBUG    init       app:__init__:557 Qt arguments: ['--enable-features=WebRTCPipeWireCapturer,OverlayScrollbar,ReducedReferrerGranularity']
20:37:50 DEBUG    init       app:__init__:562 Initializing application...
```

I haven't double checked in the code yet, but I believe there isn't much qutebrowser is doing at that point, at least not explicitly. The former is probably mostly imports of Qt and such, the latter is creating the QApplication...

Could you do another log like `existing-process.log` (i.e. from the process launched when one is there already), but with `python3 -X importtime -m qutebrowser --basedir ...` instead please?

**Comment by Hi-Angel** [comment]
> Could you do another log like `existing-process.log` (i.e. from the process launched when one is there already), but with `python3 -X importtime -m qutebrowser --basedir ...` instead please?

Sure, done: [existing-process-python-time.log](https://github.com/qutebrowser/qutebrowser/files/7499871/existing-process2.log)

> I believe there isn't much qutebrowser is doing at that point, at least not explicitly

I believe that as well. The fact that it is only reproducible on cold FS cache makes me think time is spent waiting for Qt stuff to pop up into RAM. I presume this is exactly the reason `open_url_in_instance.sh` workaround existed.

**Comment by The-Compiler** [comment]
Yup, indeed:

```
import time: self [us] | cumulative | imported package
import time:   8593091 |    8667803 | PyQt5.QtWebEngineWidgets
import time:   1217803 |    1217803 |         PyQt5.QtWebKit
```

Just as a double check: If you run [testbrowser_webengine.py](https://github.com/qutebrowser/qutebrowser/blob/master/scripts/testbrowser/testbrowser_webengine.py) with `python3 testbrowser_webengine.py qutebrowser.org`, I'm assuming this will also take something like 10 seconds+?

**Comment by Hi-Angel** [comment]
> Just as a double check: If you run [testbrowser_webengine.py](https://github.com/qutebrowser/qutebrowser/blob/master/scripts/testbrowser/testbrowser_webengine.py) with `python3 testbrowser_webengine.py qutebrowser.org`, I'm assuming this will also take something like 10 seconds+?

Given cold FS cache, it takes around 40 seconds to load.

**Comment by The-Compiler** [comment]
Hm, interesting. Could you try the [C++ testbrowser](https://github.com/qutebrowser/qutebrowser/tree/master/scripts/testbrowser/cpp/webengine) too? With those two files, you should be able to run `qmake` and `make` to get a `testbrowser` executable. I assume it takes around the same time?

Also, if you happen to have Chromium set up, how long does that take to start if not open already?

**Comment by Hi-Angel** [comment]
> Hm, interesting. Could you try the [C++ testbrowser](https://github.com/qutebrowser/qutebrowser/tree/master/scripts/testbrowser/cpp/webengine) too? With those two files, you should be able to run `qmake` and `make` to get a `testbrowser` executable. I assume it takes around the same time?

Yep, 40+ seconds *(when a window appears and then I close it, `time` shows 53 seconds to be exact)*

> Also, if you happen to have Chromium set up, how long does that take to start if not open already?

On cold FS cache, `google-chrome` takes `1:13.46` *(one minute)*, but actually the window appears at around 40-50 seconds too, and then it shows a blank screen for some more time.

------

FWIW, I have BTRFS with ZSTD compression. The compression is supposed to make disk operations to be faster *(by converting IO load to CPU load)*, I am not sure how much it would take on an FS without compression.
